packages/knip/src/plugins/pnpm/index.ts

```diff
@@ -1,10 +1,15 @@
 import type { IsPluginEnabled, Plugin } from '../../types/config.js';
+import { _firstGlob } from '../../util/glob.js';
 
 // https://pnpm.io/pnpmfile
 
 const title = 'pnpm';
 
-const isEnabled: IsPluginEnabled = ({ manifest }) => Boolean(manifest.packageManager?.startsWith('pnpm@'));
+const isEnabled: IsPluginEnabled = async ({ cwd, manifest }) =>
+  Boolean(
+    manifest.packageManager?.startsWith('pnpm@') ||
+      (await _firstGlob({ cwd, patterns: ['pnpm-lock.yaml', 'pnpm-workspace.yaml'] }))
```

Comment:
I think `pnpm-workspace.yaml` isn't needed here? Could it ever happen that there is a `pnpm-workspace.yaml`, and neither `package.json#pnpm` nor `pnpm-lock.yaml`?

Reply:
I suppose it's highly unlikely but there _is_ the [`--no-lockfile` option](https://pnpm.io/cli/install#--no-lockfile) and people might not commit the lockfile to version control

Reply:
Alright then! I'm wary of changes here as this runs in each workspace. But I see this plugin does not have `isRootOnly` yet so that'll "compensate" (I'll add it).